controllers/directional_trading/lp_hedging.py

```diff
@@ -0,0 +1,819 @@
+                            self.hedging_amount: Decimal = Decimal(position.amount)  # type: ignore
+                            self.current_hedging_usd = (
+                                self.hedging_amount * self.current_price
+                            )
+                            self.hedging_pnl = position.unrealized_pnl
+                            self.logger().debug(
+                                f"Active order found: {position}, "
+                                f"hedging amount: {self.hedging_amount}, "
+                                f"current hedging USD: {self.current_hedging_usd}"
+                            )
+                            break
+                else:
+                    self.logger().error("No active orders found")
+
+                # self.logger().info(f"Active orders: {active_orders}")
+                self.logger().debug("Positions held: %s", self.positions_held)
+            else:
+                if self.position_summary:
+                    self.hedging_amount = Decimal(self.position_summary.amount)
+                    if self.position_summary.side == TradeType.SELL:
+                        self.hedging_amount = -self.hedging_amount
+                    self.current_hedging_usd = Decimal(
+                        self.position_summary.volume_traded_quote
+                    )
+                    self.hedging_pnl = self.position_summary.unrealized_pnl_quote
+            self.total_value_usd = self.base_value_usd + self.cur_quote_token_amount
+            if self.initial_total_value_usd is None:
+                self.initial_total_value_usd = self.total_value_usd
+
+            if self.base_value_usd <= 0:
+                self.logger().error("Base value is 0, check gateway and oracle")
+                return
+
+            self.profitability = self.total_value_usd - self.initial_total_value_usd
+
+            if not self.config.is_pool_amm and not self.is_pool_in_range():
+                self.logger().warning("Pool is out of range")
+                self.processed_data["signal"] = 0
+                self.processed_data["hedge_amount"] = 0
+                return
+
+            # check if we fulfill the rebalance condition
+            self.divergence_pct = (
+                abs(self.current_hedging_usd) - self.base_value_usd
+            ) / self.base_value_usd
+
+            abs_amount_diff = (
+                abs(self.cur_base_token_amount - abs(self.hedging_amount))
+                if self.hedging_amount < 0
+                else self.hedging_amount + self.cur_base_token_amount
+            )
+
+            self.calculated_margin = (
+                abs(self.current_hedging_usd) / self.config.leverage
+            )
+            self.total_portfolio_value = (
+                self.total_value_usd
+                + self.hedging_pnl
+                + self.calculated_margin
+                - self.cumulative_fees
```

Comment:
I suspect fees are already reflected in `total_value_usd` if this calculation already uses the token amounts in the pools

Reply:
Also, I wasn't sure about the computation of `self.hedging_pnl` and `self.calculated_margin` here. Combined, this should be the P&L of the short position.